src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -1288,11 +1303,13 @@ CHIP_ERROR BLEManagerImpl::HandleGAPDisconnect(struct ble_gap_event * gapEvent)
             disconReason = BLE_ERROR_CHIPOBLE_PROTOCOL_ABORT;
             break;
         }
-        HandleConnectionError(gapEvent->disconnect.conn.conn_handle, disconReason);
+
+        // Set the connection handle if unsubscribe is successful. This connection handle is then used in
+        // _OnPlatformEvent() and HandleConnectionError() is called if its valid.
+        disconnectEvent.CHIPoBLEConnectionError.ConId  = gapEvent->disconnect.conn.conn_handle;
```

Comment:
Is there a reason this using a kCHIPoBLEConnectionClosed event instead of a kCHIPoBLEConnectionError event?  Because if we're going to examine `event->CHIPoBLEConnectionError`, it seems like we should be using a kCHIPoBLEConnectionError event (though this is not very clearly documented).

Reply:
Since this gets called after BLE disconnection, it is a connection closed event and I think that is required. But, in above case, thats an connection error, so sending `kCHIPoBLEConnectionError` rather than executing in nimble threads context.

Reply:
Then something needs to clearly define that CHIPoBLEConnectionError will be initialized for closed events, not just for error events, and how it will be initialized.

Reply:
I've added the comment there in the code about why we are calling the connection error event.

I've tested the scenario where central disconnects without unsubscribing and we do hit this code path.

Reply:
Again: the definition of CHIPoBLEConnectionError needs to have comments explaining when that event will be generated and what fields it will have in various cases.  @shubhamdp

Reply:
Added a comment where `kCHIPoBLEConnectionError` is defined.